### DISCUSSION TITLE
CoAP Device Service Scalability

### DISCUSSION THREAD

**Discussion by inimin** [edgexfoundry/community#82] [Q&A]
Hi There, 
I'm planning of using CoAP device service and also developing CBOR data type support for it, but want to know if there is any scalability solution for using this device service with more than 10,000-30,000 things?

**Comment by iain-anderson** [comment]
Hi, I'm not aware of anyone who has attempted to deploy the service at that scale. I don't think there should be issues with the software at high device cardinality but populating core-metadata with all those things would require some automation
